datalad/core/local/create.py

```diff
@@ -206,6 +209,9 @@ def __call__(
         # we know that we need to create a dataset at `path`
         assert(path is not None)
 
+        # assure cfg_proc is a list (relevant if used via Python API)
+        cfg_proc = assure_list(cfg_proc)
```

Comment:
I guess in the long run we should implement generic `assure_list` for any arg which has `action='append',`

Reply:
although, it should not be just `assure_list` but rather allow having default `None` to discriminate between "nothing was specified" and "empty list was specified". 
Then change below removing `or []` would need to be reverted.

Reply:
Yeah, automatically handling `action='append'` arguments seems like a good idea.

If backwards-compatibility weren't a concern, I'd prefer to get rid of all the `atom -> list` argument conversion that happens when using datalad's python api.  But that ship has sailed (and I don't think others share my view).